File: lottery_backend/lottery/zones.py
_LEGACY_LABELS = {"red": "红球", "blue": "蓝球"}


def get_zones(rule_config):
    """返回标准化区列表 [{key,label,min,max,count,...}]。
    新格式直读 rule_config['zones']；老格式(red/blue 顶层键)自动转换。"""
    zones = rule_config.get("zones")
    if isinstance(zones, list):
        return zones
    result = []
    for key in ("red", "blue"):
        rule = rule_config.get(key)
        if rule is None:
            continue
        zone = dict(rule)
        zone["key"] = key
        zone.setdefault("label", _LEGACY_LABELS[key])
        result.append(zone)
    return result

File: lottery_frontend/src/utils/zones.js
const KEY_LABELS = { red: '红球', blue: '蓝球', main: '号码' }
const KEY_COLORS = { red: '#e53935', blue: '#1e88e5', main: '#fb8c00' }

export function getZones(ruleConfig) {
  if (!ruleConfig) return []
  if (Array.isArray(ruleConfig.zones)) return ruleConfig.zones
  const zones = []
  for (const key of ['red', 'blue']) {
    const r = ruleConfig[key]
    if (!r) continue
    zones.push({ key, label: KEY_LABELS[key], color: KEY_COLORS[key], ...r })
  }
  return zones
}

export function zoneLabel(key) {
  return KEY_LABELS[key] || key
}

export function zoneColor(key) {
  return KEY_COLORS[key] || '#9e9e9e'
}
